emitter/file.go

```diff
@@ -33,20 +34,39 @@ func NewFile(path string, flags *zio.WriterFlags) (*zio.Writer, error) {
 	return NewFileWithSource(uri, flags, src)
 }
 
+func isTerminal(w io.Writer) bool {
+	if f, ok := w.(*os.File); ok {
+		if terminal.IsTerminal(int(f.Fd())) {
+			return true
+		}
+	}
+	return false
+}
+
 func NewFileWithSource(path iosrc.URI, flags *zio.WriterFlags, source iosrc.Source) (*zio.Writer, error) {
 	f, err := source.NewWriter(path)
 	if err != nil {
 		return nil, err
 	}
+
+	var wc io.WriteCloser
 	if path.Scheme == "stdio" {
-		// Don't close stdout in case we live inside something
-		// here that runs multiple instances of this to stdout.
-		f = &noClose{os.Stdout}
+		// Don't close stdio in case we live inside something
+		// that has multiple stdio users.
+		nc := &noClose{f}
+		if isTerminal(f) {
+			// Don't buffer terminal output.
+			wc = &noClose{f}
+		} else {
+			wc = bufwriter.New(nc)
+		}
```

Comment:
```suggestion
		wc = &noClose{f}
		if !isTerminal(f) {
			// Don't buffer terminal output.
			wc = bufwriter.New(nc)
		}
```

Reply:
Maybe do this for any terminal?
```go
	f, err := source.NewWriter(path)
	if err != nil {
		return nil, err
	}
	wc := io.WriteCloser(f)
	if !isTerminal(f) {
		wc = bufwriter.New(f)
	}
	if path.Scheme == "stdio" {
		// Don't close stdio in case we live inside something
		// that has multiple stdio users.
		wc = &noClose{wc}
	}
```